### ISSUE TITLE
no trace found for 2147483647 in default~hdd

### ISSUE THREAD

**Issue by lhns** [TheJJ/ceph-balancer#29]
Hi! I have used this balancer to great success on my cluster. Lately I reformatted some osds and moved their db to an nvme device. Since then I get the following error:
`./placementoptimizer.py -v balance --max-pg-moves 20 --max-move-attempts=20 | tee balance-upmaps`
```
Traceback (most recent call last):
  File "/home/lolhens/./placementoptimizer.py", line 5060, in <module>
    exit(main())
  File "/home/lolhens/./placementoptimizer.py", line 5028, in main
    balance(args, state)
  File "/home/lolhens/./placementoptimizer.py", line 4393, in balance
    try_pg_move.prepare_crush_check()
  File "/home/lolhens/./placementoptimizer.py", line 2496, in prepare_crush_check
    raise RuntimeError(f"no trace found for {pg_osd} in {current_root_name}")
RuntimeError: no trace found for 2147483647 in default~hdd
```

I modified the script a bit to print out the [root_osds](https://github.com/TheJJ/ceph-balancer/blob/master/placementoptimizer.py#L2488) and I get this:
`[20, 8, 2147483647, 11]`

Sadly I don't know the ceph internals enough to figure out what is going on. I saw that you did quite a few changes today and tried it with the new version but I get the same error.

**Comment by TheJJ** [comment]
Can you generate a state dump, upload it somewhere and send the link to me at jj at sft.lol?
Then I can reproduce locally and can fix it :)

**Comment by lhns** [comment]
I reformatted some osds and now it is working again. Maybe the cluster was just in a weird state.

**Comment by TheJJ** [comment]
fixed in 9feda6d9fb1aba3b70a819c70a7a4ef0e5254818

**Comment by TheJJ** [comment]
I wonder why your cluster even had the idea to put `-1` in the up set of a pg - i thought the up set always contains a valid target position, but it seems there was none available.
